pyproject.toml

```diff
@@ -0,0 +1,108 @@
+[tool.poetry]
+name = "snuba"
+version = "0.0.0"
+description = ""
+readme = "README.md"
+repository = "https://github.com/getsentry/snuba"
+license = "BSL-1.1"
+authors = []
+
+[tool.poetry.dependencies]
+python = "~3.7"
+appnope = "=0.1.0"
+argh = "=0.26.2"
+astroid = "=1.6.5"
+atomicwrites = "=1.1.5"
+attrs = "=18.1.0"
+autopep8 = "=1.3.5"
+backcall = "=0.1.0"
+black = "=19.10b0"
+blinker = "=1.4"
+certifi = "=2018.4.16"
+chardet = "=3.0.4"
+click = "=6.7"
+clickhouse-driver = "=0.1.1"
+colorama = "=0.3.9"
+configparser = "=3.5.0"
+confluent-kafka = "=1.2.0"
+contextlib2 = "=0.5.5"
+coverage = "=4.5.1"
+datadog = "=0.21.0"
+decorator = "=4.3.0"
+deprecation = "=2.0.3"
+docopt = "=0.6.2"
+enum34 = "=1.1.6"
+flake8 = "=3.5.0"
+Flask = "=1.0.2"
+funcsigs = "=1.0.2"
+future = "=0.16.0"
+honcho = "=1.0.1"
+idna = "=2.7"
+ipdb = "=0.11"
+ipython = "=6.4.0"
+ipython-genutils = "=0.2.0"
+isodate = "=0.6.0"
+isort = "=4.3.4"
+itsdangerous = "=0.24"
+jedi = "=0.12.0"
+Jinja2 = "=2.10"
+jsonschema = "=3.0.1"
+lazy-object-proxy = "=1.3.1"
+linecache2 = "=1.0.0"
+lz4 = "=2.0.0"
+Markdown = "=2.6.11"
+MarkupSafe = "=1.0"
+mccabe = "=0.6.1"
+more-itertools = "=4.2.0"
+packaging = "=17.1"
+parso = "=0.2.1"
+pathlib2 = "=2.3.2"
+pathtools = "=0.1.2"
+pbr = "=4.0.4"
+petname = "=2.2"
+pexpect = "=4.6.0"
+pickleshare = "=0.7.4"
+prompt-toolkit = "=1.0.15"
+ptyprocess = "=0.5.2"
+py = "=1.5.3"
+pycodestyle = "=2.3.1"
+pyflakes = "=1.6.0"
+Pygments = "=2.2.0"
+pylint = "=1.9.2"
+pyparsing = "=2.2.0"
+pytest = { version = "=5.2.4", optional = true }
+pytest-cov = { version = "=2.5.1", optional = true }
+pytest-watch = { version = "=4.2.0", optional = true }
+pytest-xdist = { version = "=1.30.0", optional = true }
+python-dateutil = "=2.7.3"
+pytz = "=2018.4"
+python-rapidjson = "=0.8.0"
+redis = "=2.10.6"
+redis-py-cluster = "=1.3.5"
+sentry-sdk = "=0.13.1"
+simplegeneric = "=0.8.1"
+simplejson = "=3.15.0"
+singledispatch = "=3.4.0.3"
+traceback2 = "=1.4.0"
+traitlets = "=4.3.2"
+unittest2 = "=1.1.0"
+urllib3 = "=1.25.3"
+uWSGI = "=2.0.17"
+wcwidth = "=0.1.7"
+Werkzeug = "=0.14.1"
+wrapt = "=1.10.11"
+
+[tool.poetry.extras]
+test = [
+    "pytest",
+    "pytest-cov",
+    "pytest-watch",
+    "pytest-xdist",
```

Comment:
aside i didnt realize pytest-watch was here, so we could remove pytest-xdist config since it duplicates that functionality

only thing id note is we use pytest-xdist everywhere else for 'loop on fail' (pytest -f)